tests/advanced_visibility.go

```diff
@@ -154,2499 +140,2589 @@ func (s *AdvancedVisibilitySuite) SetupTest() {
-		Namespace:          s.namespace,
-		Identity:           id,
-		WorkerVersionStamp: &commonpb.WorkerVersionStamp{BuildId: "1.1"},
-		TaskToken:          task.TaskToken,
-		Commands: []*commandpb.Command{{
-			CommandType: enumspb.COMMAND_TYPE_CONTINUE_AS_NEW_WORKFLOW_EXECUTION,
-			Attributes: &commandpb.Command_ContinueAsNewWorkflowExecutionCommandAttributes{
-				ContinueAsNewWorkflowExecutionCommandAttributes: &commandpb.ContinueAsNewWorkflowExecutionCommandAttributes{
-					WorkflowType: task.WorkflowType,
-					TaskQueue:    request.TaskQueue,
-				},
-			},
-		}},
-	})
-	s.NoError(err)
-
-	buildIDs = s.getBuildIds(ctx, task.WorkflowExecution)
-	s.Equal([]string{
-		worker_versioning.UnversionedSearchAttribute,
-		worker_versioning.UnversionedBuildIdSearchAttribute("1.0"),
-		worker_versioning.UnversionedBuildIdSearchAttribute("1.1"),
-	}, buildIDs)
-
-	task, err = s.client.PollWorkflowTaskQueue(ctx, pollRequest)
-	s.NoError(err)
-	s.Greater(len(task.TaskToken), 0)
-
-	buildIDs = s.getBuildIds(ctx, task.WorkflowExecution)
-	s.Equal([]string{}, buildIDs)
-
-	_, err = s.client.RespondWorkflowTaskCompleted(ctx, &workflowservice.RespondWorkflowTaskCompletedRequest{
-		Namespace:          s.namespace,
-		Identity:           id,
-		WorkerVersionStamp: &commonpb.WorkerVersionStamp{BuildId: "1.2"},
-		TaskToken:          task.TaskToken,
-		Commands: []*commandpb.Command{{
-			CommandType: enumspb.COMMAND_TYPE_COMPLETE_WORKFLOW_EXECUTION,
-			Attributes: &commandpb.Command_CompleteWorkflowExecutionCommandAttributes{
-				CompleteWorkflowExecutionCommandAttributes: &commandpb.CompleteWorkflowExecutionCommandAttributes{},
-			},
-		}},
-	})
-	s.NoError(err)
-
-	buildIDs = s.getBuildIds(ctx, task.WorkflowExecution)
-	s.Equal([]string{worker_versioning.UnversionedSearchAttribute, worker_versioning.UnversionedBuildIdSearchAttribute("1.2")}, buildIDs)
-
-	for minor := 1; minor <= 2; minor++ {
-		s.Eventually(func() bool {
-			response, err := s.client.ListWorkflowExecutions(ctx, &workflowservice.ListWorkflowExecutionsRequest{
-				Namespace: s.namespace,
-				Query:     fmt.Sprintf("BuildIds = '%s'", worker_versioning.UnversionedBuildIdSearchAttribute(fmt.Sprintf("1.%d", minor))),
-				PageSize:  defaultPageSize,
-			})
-			if err != nil {
-				return false
-			}
-			if len(response.Executions) == 0 {
-				return false
-			}
-			s.Equal(id, response.Executions[0].Execution.WorkflowId)
-			return true
-		}, 10*time.Second, 100*time.Millisecond)
-	}
-}
-
-func (s *AdvancedVisibilitySuite) Test_BuildIdIndexedOnCompletion_VersionedWorker() {
-	// Use only one partition to avoid having to wait for user data propagation later
-	dc := s.testCluster.host.dcClient
-	dc.OverrideValue(s.T(), dynamicconfig.MatchingNumTaskqueueReadPartitions, 1)
-	dc.OverrideValue(s.T(), dynamicconfig.MatchingNumTaskqueueWritePartitions, 1)
-
+//func (s *AdvancedVisibilitySuite) TestListOpenWorkflow() {
```

Comment:
can you undo this commenting before the review pass? it's making this diff really hard to read.

also, do you think the new tests should go somewhere around here: https://github.com/temporalio/temporal/blob/main/tests/schedule.go#L355
there are already a couple of tests using ListSchedules with a Query

Reply:
Fixed the commenting. Thanks.

As for the tests, I can see an argument on both sides. They newly added tests are testing visibility, but they are testing the visibility of schedules. At this point, I'm agnostic to either location. If you think they should move, let me know and I'll move them.